src/main/java/gg/uhc/uhc/modules/ModuleRegistry.java

```diff
@@ -86,6 +87,7 @@ public ModuleRegistry(Plugin plugin, MessageTemplates strings, ConfigurationSect
         this.strings = strings;
 
         this.addonInventory = new IconInventory(ADDON_INVENTORY_TITLE);
+        ShutdownListenerRegistry.addListener(addonInventory);
```

Comment:
Can IconInventory not just register the listener itself? Doesn't make much sense to make the module registry do it